### PR TITLE
e2e: move funs of framework/gpu to e2e_node

### PR DESCRIPTION
**Pull request by haosdent**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:

Move funs of framework/gpu to e2e_node

**Which issue(s) this PR fixes**:

Ref #86052

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: test/e2e_node/gpu_device_plugin_test.go [hunk 1]
```diff
@@ -21,18 +21,48 @@ import (
 	"strconv"
 	"time"
 
+	"k8s.io/kubernetes/test/e2e/framework/gpu"
```
Comment (oomichi) [inline test/e2e_node/gpu_device_plugin_test.go]: You don't need to move this line from line 28 at all.

Comment (haosdent) [reply test/e2e_node/gpu_device_plugin_test.go]: @oomichi thanks a lot! fixed, could you check again?

**Comment by oomichi** [comment]
Thanks for updating, let's check the test result.

/ok-to-test

**Comment by haosdent** [comment]
@oomichi @SataQiu could you help to review again? Thank you very much!

**Review by oomichi — COMMENTED** [review]
/lgtm

**Comment by tanjunchen** [comment]
/lgtm
Thanks

**Pull request merged**